astroplan/constraints.py

```diff
@@ -506,15 +506,35 @@ def is_always_observable(constraints, observer, targets, times=None,
         List of booleans of same length as ``targets`` for whether or not each
         target is observable in the time range given the constraints.
     """
+    targets_is_scalar = False
+    times_is_scalar = False
+
+    if not hasattr(targets, '__len__'):
+        targets = [targets]
+        targets_is_scalar = True
+
     if not hasattr(constraints, '__len__'):
         constraints = [constraints]
 
+    if not hasattr(times, '__len__') or not hasattr(time_range, '__len__'):
+        times_is_scalar = True
+
     applied_constraints = [constraint(observer, targets, times=times,
                                       time_range=time_range,
                                       time_grid_resolution=time_grid_resolution)
                            for constraint in constraints]
     contraint_arr = np.logical_and.reduce(applied_constraints)
-    return np.all(contraint_arr, axis=1)
+
+    always_observable = np.all(contraint_arr, axis=1)
+
+    if targets_is_scalar and times_is_scalar:
+        return always_observable.ravel()[0]
+
+    elif targets_is_scalar:
+        return always_observable[:, 0]
```

Comment:
what about the `times_is_scalar` but  not `targets_is_scalar` case?  Don't you then want to do ``always_observable[0]`` ? Or does that case broadcast to 1-d already?